### PR TITLE
MINIFICPP-1148 ClientSocket: fix addrinfo lifetime

### PR DESCRIPTION
**Pull request by szaszm**
Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [x] Does your PR title start with MINIFICPP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [x] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [x] If applicable, have you updated the LICENSE file?
- [x] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [x] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by szaszm** [comment]
With this change, SocketTests will no longer randomly fail on Windows and minifi will no longer randomly crash when a ClientSocket-based connection repeatedly fails.
I'd prefer to handle other problems with the class after this and MINIFICPP-1119 (#713) are merged.

File: libminifi/src/io/win/ClientSocket.cpp [hunk 1]
```diff
@@ -242,11 +237,17 @@ int16_t Socket::initialize() {
   hints.ai_protocol = 0; /* any protocol */
 
   int errcode = getaddrinfo(requested_hostname_.c_str(), 0, &hints, &addr_info_);
```
Comment (szaszm) [inline libminifi/src/io/win/ClientSocket.cpp]: The lifetime of `addr_info_` was changed from "from initialization to either the first failing `connect` or the end of `this`" to "during initialization".

Comment (arpadboda) [reply libminifi/src/io/win/ClientSocket.cpp]: That's great, but in that case I think it just needs to be a local variable and not a member.

Comment (szaszm) [reply libminifi/src/io/win/ClientSocket.cpp]: The value is accessed by `createConnection` as a data member and I didn't want to risk breaking anything during this bugfix by changing the signature to take an additional argument.

I'd prefer to later rewrite the whole class, that will most likely eliminate the need to pass the result of `getaddrinfo` to the function.

I can do additional investigation about the required effort if you insist on this change.

Comment (szaszm) [reply libminifi/src/io/win/ClientSocket.cpp]: done in 6512bec. Please excuse my confusion while writing the previous comment

**Comment by arpadboda** [comment]
Nice catch!

Please fix the linter errors:
```
/Users/travis/build/apache/nifi-minifi-cpp/libminifi/src//io/win/ClientSocket.cpp:41:  Do not indent within a namespace  [runtime/indentation_namespace] [4]
/Users/travis/build/apache/nifi-minifi-cpp/libminifi/src//io/win/ClientSocket.cpp:46:  Anonymous namespace should be terminated with "// namespace"  [readability/namespace] [5]
```

Otherwise I'm happy with this.

**Pull request closed without merging**